src/__experimental__/components/decimal/decimal.spec.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import TestUtils from 'react-dom/test-utils';
+import Decimal from './decimal.component';
+import Textbox from '../textbox/textbox.component';
+
+function render(props) {
+  const onChange = props.onChange ? props.onChange : () => true;
+
+  return shallow(
+    <Decimal
+      onChange={ onChange }
+      value={ props.value }
+      precision={ props.precision }
+    />
+  );
+}
+
+function renderIntoDocument(props) {
+  return TestUtils.renderIntoDocument(
+    <Decimal value={ props.value } precision={ props.precision } />
+  );
+}
+
+describe('Decimal', () => {
+  describe('Input validation', () => {
+    it('renders the correct value', () => {
+      const input = render({ value: '9.87' });
+      expect(input.find(Textbox).prop('value')).toEqual('9.87');
+    });
+
+    it('renders a value from props exactly even when invalid', () => {
```

Comment:
There's a whole bunch of formatting that happens when a value is entered (removing unwanted characters, putting the delimiters in the right place, ensuring that the precision is correct). This only happens on `blur` though (as per existing component - constantly formatting the number is quite expensive).

There's also the possibility of the component receiving an invalid value via props. As per @andrewjtait , I've written the component so that invalid values are shown in all their glory, rather than giving an inaccurate indication of the value that'd been provided.